src/acat/azext_acat/custom.py

```diff
@@ -0,0 +1,537 @@
+        args_schema.status._required = False
+        args_schema.status._registered = False
+        args_schema.update_webhook_key._required = False
+        args_schema.update_webhook_key._registered = False
+        args_schema.webhook_key._required = False
+        args_schema.webhook_key._registered = False
+
+        args_schema.content_type_with_default = AAZStrArg(
+            options=["--content-type"],
+            arg_group="Properties",
+            help="content type",
+            required=False,
+            enum={"application/json": "application/json"},
+            default="application/json",
+        )
+        args_schema.events_with_default = AAZListArg(
+            options=["--events"],
+            arg_group="Properties",
+            help="under which event notification should be sent.",
+            required=False,
+            default=[],
+        )
+        args_schema.events_with_default.Element = AAZStrArg()
+        args_schema.payload_url_required = AAZStrArg(
+            options=["--payload-url"],
+            arg_group="Properties",
+            help="webhook payload url",
+            required=True,
+            fmt=AAZStrArgFormat(
+                pattern=r"^(http(s)?://)[\S]{0,64994}$",
+            ),
+        )
+        args_schema.trigger_mode = AAZStrArg(
+            options=["--trigger-mode"],
+            arg_group="Properties",
+            help="whether to send notification under any event.",
+            default="all",
+            enum={"all": "true", "customize": "false"},
+        )
+
+        args_schema.status_with_default = AAZStrArg(
+            options=["--disable"],
+            arg_group="Properties",
+            help="Webhook status.",
+            enum={"false": "enable", "true": "disable"},
+            default="enalbe",
+            blank="disable",
+        )
+        args_schema.webhook_key_with_default = AAZStrArg(
+            options=["--secret"],
+            arg_group="Properties",
+            help="webhook secret token. If not set, this field value is null; otherwise, please set a string value.",
+            default="",
+            blank="",
+            fmt=AAZStrArgFormat(
+                pattern="^.{0,2048}$",
+            ),
+        )
+        return args_schema
+
+    def pre_operations(self):
+        from azure.cli.core.aaz.utils import assign_aaz_list_arg
+
+        args = self.ctx.args
+        args.content_type = args.content_type_with_default
+        args.events = assign_aaz_list_arg(args.events, args.events_with_default)
+        args.payload_url = args.payload_url_required
+        args.webhook_key = args.webhook_key_with_default
+        args.status = args.status_with_default
+        args.send_all_events = args.trigger_mode
+
+        wKey = str(args.webhook_key)
+        hasValidWebhookKey = wKey != "Undefined" and len(wKey) > 0
+        args.update_webhook_key = "true" if hasValidWebhookKey else "false"
```

Comment:
I think it seems the webhook_key_with_default is not required in your case. You don't need to provide the default value in argument definition, instead you can set the blank value in the pre_operations hook
```suggestion
        If has_value(args.webhook_key):
            args.update_webhook_key = "true"
        else:
            args.webhook_key = ""
            args.update_webhook_key = "false"
```